### ISSUE TITLE
Marker Not Showing

### ISSUE THREAD

**Issue by Akash187** [ordazgustavo/here-maps-react#8]
Hi, My Map is working well but the marker is not showing. Even the example is not working.

**Comment by ordazgustavo** [comment]
Hi, I'll take a look.

Can you send a code sandbox or a snippet?

Sorry for the late response.

**Comment by Akash187** [comment]
Here is the gist of code [https://gist.github.com/Akash187/b2bb659651bd8765c27a6320a461055d](url)

**Comment by ryanbaer** [comment]
I was running into a similar issue. I forked your repo and was playing with this a bit.

I believe the issue might be that `Marker:componentDidMount` is called while `map` is still `null`.

When `map` is finally set on the context, `Marker` has already mounted, so `addMarkerToMap` never gets called.

In my fork I was able to fix it. I did this by making a change within `HEREMap`, where `children` isn't rendered until `map` is set.

Hope this is helpful.

---

P.S. I wanted to suggest to switch away from the old React Context API at some point.

Here's what it would look like for `Marker`, where `this.props.mapContext` (or whatever you'd like to call it) replaces all instances of `this.context`.
```
export default props => (
  <MapContext.Consumer>
    {mapContext => <Marker mapContext={mapContext} {...props} />}
  </MapContext.Consumer>
);
```

---

P.P.S. Thanks for all of your hard work in creating and maintaining this package!
